io.asgardio.java.oidc.sdk/src/main/java/io/asgardio/java/oidc/sdk/OIDCManagerImpl.java

```diff
@@ -0,0 +1,307 @@
+                logger.log(Level.INFO, "Handling the OIDC Authorization response.");
+                boolean isAuthenticated = handleAuthentication(request, authenticationInfo);
+                if (isAuthenticated) {
+                    logger.log(Level.INFO, "Authentication successful. Redirecting to the target page.");
+                    return authenticationInfo;
+                } else {
+                    logger.log(Level.ERROR, "Authentication failed. Invalidating the session.");
+                    throw new SSOAgentServerException(
+                            SSOAgentConstants.ErrorMessages.AUTHENTICATION_FAILED.getMessage(),
+                            SSOAgentConstants.ErrorMessages.AUTHENTICATION_FAILED.getCode());
+                }
+
+            } else {
+                logger.log(Level.INFO, "Clearing the active session and redirecting.");
+                throw new SSOAgentServerException(SSOAgentConstants.ErrorMessages.AUTHENTICATION_FAILED.getMessage(),
+                        SSOAgentConstants.ErrorMessages.AUTHENTICATION_FAILED.getCode());
+            }
+        } catch (SSOAgentServerException e) {
+            throw new SSOAgentException(e.getMessage(), e.getErrorCode());
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void logout(AuthenticationInfo authenticationInfo, HttpServletResponse response, String state)
+            throws SSOAgentException {
+
+        if (oidcAgentConfig.getPostLogoutRedirectURI() == null) {
+            logger.info("postLogoutRedirectURI is not configured. Using the callbackURL instead.");
+            URI callbackURI = oidcAgentConfig.getCallbackUrl();
+            oidcAgentConfig.setPostLogoutRedirectURI(callbackURI);
+        }
+        OIDCRequestBuilder requestBuilder = new OIDCRequestBuilder(oidcAgentConfig);
+        String logoutRequest = requestBuilder.buildLogoutRequest(authenticationInfo, state);
+        try {
+            response.sendRedirect(logoutRequest);
+        } catch (IOException e) {
+            throw new SSOAgentException(SSOAgentConstants.ErrorMessages.SERVLET_CONNECTION.getMessage(),
+                    SSOAgentConstants.ErrorMessages.SERVLET_CONNECTION.getCode(), e);
+        }
+    }
+
+    private boolean handleAuthentication(final HttpServletRequest request, AuthenticationInfo authenticationInfo) {
+
+        AuthorizationResponse authorizationResponse;
+        AuthorizationCode authorizationCode;
+        AuthorizationSuccessResponse successResponse;
+        TokenRequest tokenRequest;
+        TokenResponse tokenResponse;
+
+        try {
+            authorizationResponse = AuthorizationResponse.parse(ServletUtils.createHTTPRequest(request));
+
+            if (!authorizationResponse.indicatesSuccess()) {
+                handleErrorAuthorizationResponse(authorizationResponse);
+                return false;
+            } else {
```

Comment:
Do we actually need to have an else statement in the default flow?

Reply:
Resolved with https://github.com/asgardio/asgardio-java-oidc-sdk/pull/1/commits/87be1ad8b44e103c81d6ce8a6ed22d4f6fc4f854.